dotnet/CMakeLists.txt

```diff
@@ -0,0 +1,67 @@
+cmake_minimum_required(VERSION 2.8.11)
+
+include_directories(..)
+#Find the packages and libraries
+IF(WIN32)
+    SET(FIND_LIBRARY_USE_LIB64_PATHS ON)
+    SET(ENGINE_ARCH "x64")
+ENDIF(WIN32)
+
+IF(UNIX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__FILENAME__='\"$(subst ${CMAKE_SOURCE_DIR}/,,$(abspath $<))\"'")
+ENDIF(UNIX)
+IF(WIN32)
+    #TODO find a way to define the __FILENAME__ in Windows so that the debug output doesn't contain the absolute path.
+    #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__FILENAME__=\"$(notdir $<)\"")
+ENDIF(WIN32)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBUILDING_MYMCSAPI")
+
+set(DOTNET_AVAILABLE 0)
+
+execute_process(COMMAND dotnet --version OUTPUT_VARIABLE DOTNET_VERSION_AVAILABLE OUTPUT_STRIP_TRAILING_WHITESPACE)
+IF(DOTNET_VERSION_AVAILABLE VERSION_GREATER 2.0)
+    set(DOTNET_AVAILABLE 1)
+    MESSAGE(STATUS "dotnet ${DOTNET_VERSION_AVAILABLE} Found")
+ENDIF(DOTNET_VERSION_AVAILABLE VERSION_GREATER 2.0)
+
+IF(NOT DOTNET_AVAILABLE)
+    MESSAGE(FATAL_ERROR "dotnet command line not available")
+ENDIF()
+
```

Comment:
If not integrated into the continuous integration pipeline you have to add additional compile flags similar to the ones in the top level CMakeLists.txt to be compile-able on CentOS 7 which uses an older version of g++.

```
# Disable no-deprecated-declarations
set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG")
if(CMAKE_COMPILER_IS_GNUCXX)
    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Wall -Wunused -Wwrite-strings -Wno-strict-aliasing -Wextra -Werror -Wno-deprecated-declarations")
endif(CMAKE_COMPILER_IS_GNUCXX)

# Disable format-truncation since this triggers in mcsapi_types for something that isn't a problem
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
if(CMAKE_COMPILER_IS_GNUCXX)
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wformat -Wformat-security -fstack-protector-all -fstack-check -Wno-format-truncation -Wno-unused-command-line-argument -Wno-unknown-warning-option -pie -fPIC")
endif(CMAKE_COMPILER_IS_GNUCXX)
```